src/transformers/integrations.py

```diff
@@ -676,24 +697,56 @@ def setup(self, args, state, model, **kwargs):
         variables:
 
         Environment:
-        - **WANDB_LOG_MODEL** (`str`, *optional*, defaults to `"false"`):
-            Whether to log model and checkpoints during training. Can be `"end"`, `"checkpoint"` or `"false"`. If set
-            to `"end"`, the model will be uploaded at the end of training. If set to `"checkpoint"`, the checkpoint
-            will be uploaded every `args.save_steps` . If set to `"false"`, the model will not be uploaded. Use along
-            with [`~transformers.TrainingArguments.load_best_model_at_end`] to upload best model.
-
-            <Deprecated version="5.0">
-
-            Setting `WANDB_LOG_MODEL` as `bool` will be deprecated in version 5 of 🤗 Transformers.
-
-            </Deprecated>
-        - **WANDB_WATCH** (`str`, *optional* defaults to `"false"`):
-            Can be `"gradients"`, `"all"`, `"parameters"`, or `"false"`. Set to `"all"` to log gradients and
-            parameters.
-        - **WANDB_PROJECT** (`str`, *optional*, defaults to `"huggingface"`):
-            Set this to a custom string to store results in a different project.
+        - **WANDB_ANONYMOUS** (`str`, *optional*, defaults to `None`):
+            Set this to `"allow"`, `"never"`, or `"must"` to let users create anonymous runs with secret urls.
+        - **WANDB_CONFIG_DIR** (`str`, *optional*, defaults to `None`):
+            Set this to a custom string to store config in a different directory.
+            If not set, defaults to `"~/.config/wandb"`.
+        - **WANDB_DIR** (`str`, *optional*, defaults to `None`):
+            Set this to an absolute path to store all generated files here instead of the wandb directory
+            relative to your training script. Be sure this directory exists and the user your process runs can write to it.
         - **WANDB_DISABLED** (`bool`, *optional*, defaults to `False`):
-            Whether to disable wandb entirely. Set `WANDB_DISABLED=true` to disable.
+            Whether or not to disable wandb entirely. Set `WANDB_DISABLED=True` to disable.
+        - **WANDB_DISABLE_CODE** (`bool`, *optional*, defaults to `False`):
+            Whether or not to disable code saving. Set `WANDB_DISABLE_CODE=True` to disable.
+        - **WANDB_ENTITY** (`str`, *optional*, defaults to `None`):
+            The entity associated with your run. If you have run wandb init in the directory of your training script,
+            it will create a directory named wandb and will save a default entity which can be checked into source control.
+            If you don't want to create that file or want to override the file you can use the environmental variable.
+        - **WANDB_JOB_TYPE** (`str`, *optional*, defaults to `None`):
+            Specify the job type, like `"training"` or `"evaluation"` to indicate different types of runs.
+        - **WANDB_LOG_MODEL** (`bool`, *optional*, defaults to `False`):
```

Comment:
@nbroad1881 : Could you leave this option's docstring as it was earlier. - It really explains to the user what the `WANDB_LOG_MODEL` does adn the available options. Also, this has been changed to `str` from `bool`. Here's the docstring for your reference.

```
**WANDB_LOG_MODEL** (`str`, *optional*, defaults to `"false"`):
            Whether to log model and checkpoints during training. Can be `"end"`, `"checkpoint"` or `"false"`. If set
            to `"end"`, the model will be uploaded at the end of training. If set to `"checkpoint"`, the checkpoint
            will be uploaded every `args.save_steps` . If set to `"false"`, the model will not be uploaded. Use along
            with [`~transformers.TrainingArguments.load_best_model_at_end`] to upload best model.
            <Deprecated version="5.0">
            Setting `WANDB_LOG_MODEL` as `bool` will be deprecated in version 5 of 🤗 Transformers.
            </Deprecated>
```